packages/node_modules/@webex/webex-core/src/lib/services/interceptors/service.js

```diff
@@ -21,100 +21,80 @@ export default class ServiceInterceptor extends Interceptor {
+    this.normalizeOptions(options);
+    this.validateOptions(options);
 
-      if (service) {
-        this.validateOptions(options);
+    // Destructure commonly referenced namespaces.
+    const {services} = this.webex.internal;
+    const {service, resource} = options;
 
-        options.uri = this.generateUrl(service, options.resource);
+    // Attempt to collect the service url.
+    return services.waitForService({name: service})
+      .then((serviceUrl) => {
+        // Generate the combined service url and resource.
+        options.uri = this.generateUri(serviceUrl, resource);
 
         return options;
-      }
-
-      if (this.webex.credentials.canAuthorize) {
-        return services.waitForCatalog('postauth', 20)
-          .then(() => {
-            this.validateOptions(options);
-
-            service = services.get(options.service, true, options.catalog);
-            if (!service && options.uri) {
-              return options;
-            }
+      })
+      .catch(() => Promise.reject(new Error(
+        `service-interceptor: '${service}' is not a known service`
+      )));
+  }
 
-            if (!service) {
-              throw new Error(`\`${options.service}\` is not a known service`);
-            }
+  /* eslint-disable class-methods-use-this */
+  /**
+   * Generate a usable request uri string from a service url and a resouce.
+   *
+   * @param {string} serviceUrl - The service url.
+   * @param {string} resource - The resouce to be appended to the service url.
+   * @returns {string} - The combined service url and resource.
+   */
+  generateUri(serviceUrl, resource) {
+    const formattedService = serviceUrl.replace(trailingSlashes, '');
+    const formattedResource = resource.replace(trailingSlashes, '');
 
-            options.uri = this.generateUrl(service, options.resource);
+    return `${formattedService}/${formattedResource}`;
+  }
 
-            return Promise.resolve(options);
-          })
-          .catch((e) => Promise.reject(e));
-      }
+  /**
+   * Normalizes request options relative to service identification.
+   *
+   * @param {Object} options - The request PTO.
+   * @returns {Object} - The mudated request PTO.
+   */
+  normalizeOptions(options) {
+    // Validate if the api property is used.
+    if (options.api) {
+      // Assign the service property the value of the api property if necessary.
+      options.service = options.service || options.api;
+      delete options.api;
     }
-
-    return options;
   }
-  /* eslint-enable no-param-reassign */
 
-  /* eslint-disable class-methods-use-this */
   /**
-   * Verify that all required parameters have been specified.
-   * @param {object} options
-   * @param {string} options.resource
-   * @returns {void}
+   * Validates that the appropriate options for this interceptor are present.
+   *
+   * @param {Object} options - The request PTO.
+   * @returns {Object} - The mudated request PTO.
    */
   validateOptions(options) {
```

Comment:
Same as above

Reply:
Decided against moving these, as it would require a special-case export to perform unit tests against the methods. Which would be unlike all other interceptors.